addons/mrp/models/mrp_production.py

```diff
@@ -837,6 +837,12 @@ def _check_byproducts(self):
             if sum(order.move_byproduct_ids.filtered(lambda m: m.state != 'cancel').mapped('cost_share')) > 100:
                 raise ValidationError(_("The total cost share for a manufacturing order's by-products cannot exceed 100."))
 
+    @api.onchange('move_raw_ids')
+    def _onchange_move_raw_ids(self):
+        if self.state in ('progress', 'to_close'):
+            moves_to_pick = self.move_raw_ids.filtered(lambda m: not m.picked and not m.manual_consumption)
+            moves_to_pick.picked = True
```

Comment:
This makes you unable to un-picked a component once it's been set. :confused: 

Rather than using an `onchange`, it might be better to use the move `create()` override in `mrp` instead?

Reply:
Ok, Now It Fixed.